### PR TITLE
Add Cloud Storage encryption samples and tests.

### PR DESCRIPTION
**Pull request by jmdobry**
@ace-n Can you take a look?

### CODE REVIEW

File: storage/encryption.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+ * Uploads a file to a Google Cloud Storage bucket using a custom encryption key.
+ *
+ * The file will be encrypted by Google Cloud Storage and only retrievable using
+ * the provided encryption key.
+ *
+ * @param {string} bucketName The bucket where the file will be uploaded.
+ * @param {string} srcFileName The local file to be uploaded.
+ * @param {string} destFileName The name of the destination file.
+ * @param {string} key The encryption key.
+ * @param {function} key The callback function.
```
Comment (ace-n) [inline storage/encryption.js]: Did you mean "callback"?

File: storage/encryption.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+// [START download_encrypted_file]
+/**
+ * Downloads a previously-encrypted file from Google Cloud Storage.
+ *
+ * The encryption key provided must be the same key provided when uploading the
+ * file.
+ *
+ * @param {string} bucketName The bucket from which the file will be downloaded.
+ * @param {string} srcFileName The local file to be downloaded.
```
Comment (ace-n) [inline storage/encryption.js]: Is "local" the right wording here?

File: storage/system-test/encryption.test.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+var downloadFilePath = path.join(__dirname, '../resources/downloaded.txt');
+
+describe('storage:encryption', function () {
+  var key;
+  before(function (done) {
+    key = program.generateEncryptionKey();
+    storage.createBucket(bucketName, done);
+  });
+
+  after(function (done) {
```
Comment (ace-n) [inline storage/system-test/encryption.test.js]: Explain what this does with a comment? (e.g. "Reset GCS bucket after running all tests" or something.)

File: storage/test/encryption.test.js [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+  };
+}
+
+describe('storage:encryption', function () {
+  describe('generateEncryptionKey', function () {
+    it('should generate an encryption key', function () {
+      var program = getSample().program;
+
+      var key = program.generateEncryptionKey();
+      assert(console.log.calledWith('Base 64 encoded encryption key: %s', key));
```
Comment (ace-n) [inline storage/test/encryption.test.js]: Is there any way to validate the generated key? (Do we care?)

File: storage/test/encryption.test.js [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+        assert(err);
+        assert.equal(err.message, error);
+        assert.equal(file, undefined);
+      });
+    });
+  });
+
+  describe('downloadEncryptedFile', function () {
+    var fileName = 'test.txt';
+    it('should upload a file', function () {
```
Comment (ace-n) [inline storage/test/encryption.test.js]: Typo? I think it should be __download__

File: storage/test/encryption.test.js [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+      sample.program.downloadEncryptedFile(bucketName, fileName, fileName, 'key', function (err, file) {
+        assert(err);
+        assert.equal(err.message, error);
+        assert.equal(file, undefined);
+      });
+    });
+  });
+
+  describe('rotateEncryptionKey', function () {
+    it('should rotate an encryption key', function () {
```
Comment (ace-n) [inline storage/test/encryption.test.js]: May want to put an __it ('should be implemented')__ here.

**Comment by ace-n** [comment]
LGTM other than my comments.

File: storage/encryption.js (hunk 1 above)
Comment (jmdobry) [inline storage/encryption.js]: Fixed.

File: storage/encryption.js (hunk 2 above)
Comment (jmdobry) [inline storage/encryption.js]: Fixed.

File: storage/system-test/encryption.test.js (hunk 3 above)
Comment (jmdobry) [inline storage/system-test/encryption.test.js]: Added comments

File: storage/test/encryption.test.js (hunk 4 above)
Comment (jmdobry) [inline storage/test/encryption.test.js]: Don't care, for this demo the key can be anything.

File: storage/test/encryption.test.js (hunk 5 above)
Comment (jmdobry) [inline storage/test/encryption.test.js]: Fixed.

File: storage/test/encryption.test.js (hunk 6 above)
Comment (jmdobry) [inline storage/test/encryption.test.js]: Added

File: storage/system-test/encryption.test.js [hunk 7]
```diff
@@ -27,21 +27,27 @@ var downloadFilePath = path.join(__dirname, '../resources/downloaded.txt');
 
 describe('storage:encryption', function () {
   var key;
+
   before(function (done) {
+    // Create an key to use throughout the test
```
Comment (ace-n) [inline storage/system-test/encryption.test.js]: Grammar nit: create **a** key.

Comment (jmdobry) [inline storage/system-test/encryption.test.js]: Fixed

**Comment by ace-n** [comment]
LGTM other than my minor grammar nit. (Feel free to merge it and/or run it past other reviewers.)

**Pull request merged**